bmq-sdk/src/main/java/com/bloomberg/bmq/Session.java

```diff
@@ -1244,6 +1248,18 @@ public MessageGUID messageGUID() {
                 return impl.messageGUID();
             }
 
+            @Override
+            public CorrelationId correlationId() {
+                for (Integer sId : impl.subQueueIds()) {
+                    Subscription subscription =
+                            QueueAdapter.this.impl.getQueueOptions().getSubscriptions().get(sId);
```

Comment:
I suppose this should not work anymore and we should keep map of id->Subscription separately?